File: 49-group-anagrams/49-group-anagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        result = defaultdict(list)

        for str in strs:
            s = ''.join(sorted(str))
            result[s].append(str)
            
        return result.values()
        

File: 49-group-anagrams/49-group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
   const hash = {}
   const result = []
   
   for (let s of strs) {
       let letters = s.split('').sort().join('')
       hash[letters] = hash[letters] || []
       hash[letters].push(s)      
   }
    for (let value in hash) {
        result.push(hash[value])
    }
    return result    
};

